src/matvis/cpu/matprod.py

```diff
@@ -50,3 +50,41 @@ def compute(self, z: np.ndarray, out: np.ndarray) -> np.ndarray:
             out[i] = z[ai].conj().dot(z[aj].T)
 
         return out
+
+
+class CPUMatChunk(MatProd):
+    def compute(self, z: np.ndarray, out: np.ndarray) -> np.ndarray:
+        """Perform the source-summing operation for a single time and chunk.
+
+        Parameters
+        ----------
+        z
+            Complex integrand. Shape=(Nfeed, Nant, Nax, Nsrc).
+        out
+            Output array, shaped as (Nfeed, Nfeed, Npairs).
+        """
+
+        z = z.reshape((self.nant, self.nfeed, -1))
+
+        mat_product = np.zeros(
+            (self.nant, self.nant, self.nfeed, self.nfeed), dtype=z.dtype
+        )
```

Comment:
Would it not be faster to just save Nsets arrays in a list, and then index from them? This way, you have to put the results into a big array so they're very separated in memory space

Reply:
I mean, setup something like this:

```python
mat_product = [
    np.zeros(len(ai), len(aj), self.nfeed, self.nfeed) for ai, aj in self.matsets
]
```

Reply:
The reason I didn't do it this way is because I wanted final array that holds all the matrix products to be indexable with the elements of antpairs.  That way, I can easily grab the non-redundant pairs at the end of return an array with the same format that you'd get using the vector-vector method.  Otherwise, I have to re-calculate the mapping the mapping to the antpairs indices for each sub-matrix.  I can change it if you think it would help with memory management to do it the other way though.